app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
+  end
+
+
+  def has_company?
+    has_membership_application? && (membership_applications.select { |app| app.company && app.company.persisted? }).count > 0
+  end
+
+
+  def membership_application
+    has_membership_application? ? membership_applications.last : nil
+  end
+
+
+  def company
+    has_company? ? membership_application.company : nil
+  end
+
+
+  def is_member?
+    admin? || ( has_membership_application? ? membership_application.status == 'Godkänd' : false)
```

Comment:
Agreed.  Thanks for catching that.

There are places where we do need to do something (like display a particular menu section -- which is where this is used) if a user is a member _or_ admin.  It will be both cumbersome and not DRY if we have to repeat that line (`admin? || ( has_membership_application....`) in multiple places.

So how about if I rename this to something that makes it explicit?  Like
```def is_member_or_admin?```